tests/test_addc.py

```diff
@@ -0,0 +1,147 @@
+#
+# sambacc: a samba container configuration tool
+# Copyright (C) 2022  John Mulligan
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>
+#
+
+import os
+
+import pytest
+
+
+import sambacc.addc
+
+
+def _fake_samba_tool(path):
+    fake_samba_tool = path / "fake_samba_tool.sh"
+    with open(fake_samba_tool, "w") as fh:
+        fh.write("#!/bin/sh\n")
+        fh.write(f"[ -e {path}/fail ] && exit 1\n")
+        fh.write(f'echo "$@" > {path}/args.out\n')
+        fh.write("exit 0")
+    os.chmod(fake_samba_tool, 0o700)
```

Comment:
The code is OK, but as you are already using modern Python `pathlib.Path`, wouldn't it be better to use its provided methods? In this case, [write_text](https://docs.python.org/3/library/pathlib.html#pathlib.Path.write_text) and [chmod](https://docs.python.org/3/library/pathlib.html#pathlib.Path.chmod).

This comment holds for the rest of the code in this file.

Reply:
I've been writing python for a long time, much longer than pathlib's been around. While I generally like pathlib, old habits die hard. The library is well integrated with the os functions. Plus, in the case of a more complex multi-line file I don't really see write_text as an advantage.  For the test code already using pathlib, I will try to use it more. However, I don't think I want to change this PR so extensively to do so.

Reply:
Fair enough.